sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIClient.cs

```diff
@@ -339,10 +338,25 @@ private static PipelinePolicy CreateAddUserAgentHeaderPolicy(AzureOpenAIClientOp
         });
     }
 
+    private static PipelinePolicy CreateAddClientRequestIdHeaderPolicy()
+    {
+        return new GenericActionPipelinePolicy(message =>
+        {
+            if (message?.Request?.Headers is not null)
+            {
+                string requestId = message.Request.Headers.TryGetValue(s_clientRequestIdHeaderKey, out string existingHeader) == true
+                    ? existingHeader
+                    : Guid.NewGuid().ToString().ToLowerInvariant();
+                message.Request.Headers.Set(s_clientRequestIdHeaderKey, requestId);
+            }
+        });
+    }
+
     private static readonly string s_aoaiEndpointEnvironmentVariable = "AZURE_OPENAI_ENDPOINT";
     private static readonly string s_aoaiApiKeyEnvironmentVariable = "AZURE_OPENAI_API_KEY";
     private static readonly string s_userAgentHeaderKey = "User-Agent";
-    private static PipelineMessageClassifier _pipelineMessageClassifier;
+    private static readonly string s_clientRequestIdHeaderKey = "x-ms-client-request-id";
+    private static PipelineMessageClassifier s_pipelineMessageClassifier;
     internal static PipelineMessageClassifier PipelineMessageClassifier
-        => _pipelineMessageClassifier ??= PipelineMessageClassifier.Create(stackalloc ushort[] { 200, 201 });
+        => s_pipelineMessageClassifier ??= PipelineMessageClassifier.Create(stackalloc ushort[] { 200, 201 });
```

Comment:
nit: we don't tend to use this naming convention in Azure clients.  I don't feel strongly at all, but if this will eventually be generated it might be changed back just because the generator names it differently.

Reply:
Thanks for catching this! I "followed the analyzer" for that one ("gray squiggles"); if we're using mismatched analyzer settings (highly probable), we should fix that all up together for consistency. I'll leave it for now but make a note to go look into that.